odoo/osv/expression.py

```diff
@@ -711,6 +712,39 @@ def push_result(query, params):
                 subquery, subparams = query.select('"%s"."%s"' % (comodel._table, field.inverse_name))
                 push(('id', 'inselect', (subquery, subparams)), model, alias, internal=True)
 
+            elif field.type in ('properties', ):
+                assert len(path) == 2
+                assert "." not in path[1]
+                assert operator in ('=', '!=', '>', '>=', '<', '<=', 'in', 'not in', 'like', 'ilike')
+                assert regex_object_name.match(path[1]), "Wrong property name"
+
+                def _convert_json_type(value):
+                    if isinstance(value, str):
+                        return value
+                    elif isinstance(value, (int, float, bool)):
+                        return json.dumps(value)
+                    elif isinstance(right, (list, tuple)):
+                        return [_convert_json_type(r) for r in right]
+
+                    raise ValueError(f"Wrong value {value!r}.")
+
+                if 'like' in operator:
+                    right = f'%{right}%'
+
+                inverse = ''
+                if operator in ('in', 'not in'):
+                    if operator == 'not in':
```

Comment:
You add NOT at the begin but doesn't change the operator 'not in', it is wanted?

Reply:
Both `in` and `not in` use the SQL operator `@>` (just `not in` add a `NOT` to invert the condition)

```
operator = '@>'
```